src/test/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriterFlushTests.java

```diff
@@ -264,16 +271,21 @@ public void testFlush_WithQuantization() {
 
             IntStream.range(0, vectorsPerField.size()).forEach(i -> {
                 try {
-                    verify(knn990QuantWriterMockedConstruction.constructed().get(0)).writeState(i, quantizationState);
-                    verify(nativeIndexWriter).flushIndex(expectedVectorValues.get(i), vectorsPerField.get(i).size());
+                    if (vectorsPerField.get(i).isEmpty()) {
+                        verify(knn990QuantWriterMockedConstruction.constructed().get(0), never()).writeState(i, quantizationState);
+                        verify(nativeIndexWriter, never()).flushIndex(expectedVectorValues.get(i), vectorsPerField.get(i).size());
```

Comment:
same here, verifying no interactions is a much stronger check
```suggestion
                         verifyNoInteractions(nativeIndexWriter);
```

Reply:
I am not sure will that work. We have interactions from other fields. I only want to verify that it doesn't interact when there are empty fields.

Reply:
Makes sense, I missed that it has other values apart from empty vector